src/main/java/com/epam/reportportal/testng/TestNGService.java

```diff
@@ -147,10 +202,26 @@ public void startTestMethod(ITestResult testResult) {
 		if (rq == null) {
 			return;
 		}
-
 		Maybe<String> stepMaybe = launch.get().startTestItem(this.<Maybe<String>>getAttribute(testResult.getTestContext(), RP_ID), rq);
 		testResult.setAttribute(RP_ID, stepMaybe);
 		StepAspect.setParentId(stepMaybe);
+		if (launch.get().getParameters().isCallbackReportingEnabled()) {
+			addToTree(testResult, stepMaybe);
+		}
+	}
+
+	private void addToTree(ITestResult testResult, Maybe<String> stepMaybe) {
+		ITestContext testContext = testResult.getTestContext();
+		TestItemTree.TestItemLeaf suiteLeaf = ITEM_TREE.getTestItems().get(createKey(testContext.getSuite()));
+		if (suiteLeaf != null) {
+			TestItemTree.TestItemLeaf testLeaf = suiteLeaf.getChildItems().get(createKey(testContext));
+			if (testLeaf != null) {
+				TestItemTree.TestItemLeaf testClassLeaf = testLeaf.getChildItems().get(createKey(testResult.getTestClass()));
```

Comment:
what if more test classes will be under the test class and the structure is not "suite -> test -> step"?
what if "suite -> suite -> test -> step" or similar ?

Reply:
+1

Reply:
It doesn't matter what structure of suite is in the `suite.xml` file (with nested <suite-file .../>), because all nested suites reported to the same level as root suite, so "suite"->"test"->"step" is the only possible structure